### PR TITLE
[SPARK-6423][Mesos] MemoryUtils should use memoryOverhead if it's set

### PR DESCRIPTION
**Pull request by jongyoul**
- Fixed calculateTotalMemory to use spark.mesos.executor.memoryOverhead
- Added testCase

### CODE REVIEW

**Comment by jongyoul** [comment]
@srowen I made a PR for handling with unintended behaviour issued by #5065. Please review this.

**Comment by srowen** [comment]
Excellent, thank you. This looks like the right fix given our previous discussion at https://github.com/apache/spark/pull/5065#issuecomment-82244546. It makes Mesos work like YARN does here.

File: core/src/test/scala/org/apache/spark/scheduler/cluster/mesos/MemoryUtilsSuite.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler.cluster.mesos
+
+import org.apache.spark.{SparkConf, SparkContext}
+import org.mockito.Mockito._
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/scheduler/cluster/mesos/MemoryUtilsSuite.scala]: Nit: put the Spark imports after the third-party imports.

Also, doesn't the Mesos documentation need to change to reflect this slightly different logic? the same file updated in https://github.com/apache/spark/pull/5065/files

Comment (jongyoul) [inline core/src/test/scala/org/apache/spark/scheduler/cluster/mesos/MemoryUtilsSuite.scala]: @srowen Yes, I've known that I should fit import orders, but I missed it. And I will also update the description.

**Pull request closed without merging**